### PR TITLE
EES-5281 prevent deleting and replacing data for files with api data sets

### PR DESCRIPTION
**Pull request by amyb-hiveit**
Shows a modal informing the user that they can't replace data / delete data files if the files are linked to an API data set.

### CODE REVIEW

File: src/explore-education-statistics-admin/src/pages/release/data/components/__tests__/ReleaseDataUploadsSection.test.tsx [hunk 1]
```diff
@@ -461,6 +461,49 @@ describe('ReleaseDataUploadsSection', () => {
...
+          <ReleaseDataUploadsSection
+            publicationId="publication-1"
+            releaseId="release-1"
+            canUpdateRelease
+          />
+        </MemoryRouter>,
+      );
+
+      await waitFor(() => {
+        expect(screen.getAllByTestId('accordionSection')).toHaveLength(1);
```
Comment (luke-h1) [inline src/explore-education-statistics-admin/src/pages/release/data/components/__tests__/ReleaseDataUploadsSection.test.tsx:484]: Just a testing-related question - I was just wondering why you're using `getAllByTestId` here if there is only one? I can see down below that we're getting all `accordionSection` elements. Not sure if I'm missing something 🤔 or if the intention is to add more sections in the future

File: src/explore-education-statistics-admin/src/pages/release/data/components/ReleaseDataUploadsSection.tsx [hunk 2]
```diff
@@ -244,36 +245,68 @@ const ReleaseDataUploadsSection = ({
...
+                                  title="Cannot replace data"
+                                  triggerButton={
+                                    <ButtonText>Replace data</ButtonText>
+                                  }
+                                >
+                                  <p>
+                                    This data file has an API data set linked to
+                                    it. Please remove the API data set before
+                                    replacing the data.
+                                  </p>
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/release/data/components/ReleaseDataUploadsSection.tsx:260]: As discussed on Slack, it'd be good to have a link to the API data set in this message. I've pushed up a change that replaces `isLinkedToApiDataSet` with `publicApiDataSetId`. We should be able to use this link to the relevant API data set page.

In terms of the message, not entirely sure what the best way of presenting this is. Maybe we could add the link to one of the e.g. 'The data file has an [API data set](#test) linked to it. Please remove the API data set before replacing the data.'.

Alternatively, there could be a 'Go to API data set' link. Will leave it up to you!

File: src/explore-education-statistics-admin/src/pages/release/data/components/ReleaseDataUploadsSection.tsx [hunk 3]
```diff
@@ -244,36 +245,68 @@ const ReleaseDataUploadsSection = ({
...
+                              triggerButton={
+                                <ButtonText className="govuk-!-margin-left-3">
+                                  Delete files
+                                </ButtonText>
+                              }
+                            >
+                              <p>
+                                This data file has an API data set linked to it.
+                                Please remove the API data set before deleting.
+                              </p>
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/release/data/components/ReleaseDataUploadsSection.tsx:292]: As above - we can now adjust this message to link through to the API data set.

File: src/explore-education-statistics-admin/src/pages/release/data/components/ReleaseDataUploadsSection.tsx [hunk 4]
```diff
@@ -244,36 +245,68 @@ const ReleaseDataUploadsSection = ({
...
+                            <ButtonText
+                              onClick={() =>
+                                releaseDataFileService
+                                  .getDeleteDataFilePlan(releaseId, dataFile)
+                                  .then(plan => {
+                                    setDeleteDataFile({
+                                      plan,
+                                      file: dataFile,
+                                    });
+                                  })
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/release/data/components/ReleaseDataUploadsSection.tsx:304]: Pretty minor, but could async / await this?

File: src/explore-education-statistics-admin/src/services/releaseDataFileService.ts [hunk 5]
```diff
@@ -38,6 +38,7 @@ export interface DataFile {
   created?: string;
   isDeleting?: boolean;
   isCancelling?: boolean;
+  isLinkedToApiDataSet?: boolean;
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/services/releaseDataFileService.ts:41]: As per earlier comments - this has been replaced with the following:

- `publicApiDataSetId` (string) - can be used for linking
- `publicApiDataSetVersion` (string) - can't be used for linking, but added for completeness and potential future use-cases / debugging

**Review by ntsim — CHANGES_REQUESTED** [review]

File: src/explore-education-statistics-admin/src/pages/release/data/components/ReleaseDataUploadsSection.tsx (hunk 2 above)
Comment (amyb-hiveit) [reply src/explore-education-statistics-admin/src/pages/release/data/components/ReleaseDataUploadsSection.tsx:262]: I've gone with adding a  'Go to API data set' link.

**Review by ntsim — APPROVED** [review]

**Pull request merged**